src/mower_logic/src/monitoring/monitoring.cpp

```diff
@@ -178,9 +144,61 @@ void set_sensor_limits(SensorConfig &sensor_config) {
   if (sensor_config.si.upper_critical_value) sensor_config.si.has_critical_high = true;
 }
 
+void set_limits_battery_v(SensorConfig &sensor_config) {
+  sensor_config.si.lower_critical_value = mower_logic_config.battery_critical_voltage;
+  sensor_config.si.min_value = mower_logic_config.battery_empty_voltage;
+  sensor_config.si.max_value = mower_logic_config.battery_full_voltage;
+  sensor_config.si.upper_critical_value = mower_logic_config.battery_critical_high_voltage;
+  set_si_has_props(sensor_config);
+}
+
+void set_limits_charge_v(SensorConfig &sensor_config) {
+  // FIXME: Shall these better go to mower_logic's dyn-reconfigure (or an own dyn-reconfigure server)?
+  sensor_config.si.min_value = 24.0f;             // Mnimum voltage for before deep-discharge
+  sensor_config.si.max_value = 29.2f;             // Optimal charge voltage
+  sensor_config.si.upper_critical_value = 30.0f;  // Taken from OpenMower FW
+  set_si_has_props(sensor_config);
```

Comment:
Not sure if you're planning to have some functions which _don't_ call `set_si_has_props()`, but if not, you could remove the call from all the variants and instead call it below:
```c++
if (sc_pair.second.setSensorLimitsCB) {
    sc_pair.second.setSensorLimitsCB(sc_pair.second);
    set_si_has_props(sc_pair.second);
}
```

Instead of the second call, you could of course also paste the code there directly, then it would still be in a single place.